drape/glyph_manager.cpp

```diff
@@ -659,46 +626,68 @@ hb_language_t OrganicMapsLanguageToHarfbuzzLanguage(int8_t lang)
 }
 }  // namespace
 
+// This method is NOT multithreading-safe.
 text::TextMetrics GlyphManager::ShapeText(std::string_view utf8, int fontPixelHeight, int8_t lang)
 {
+#ifdef DEBUG
+  static int const fontSize = fontPixelHeight;
+  ASSERT_EQUAL(fontSize, fontPixelHeight,
+      ("Cache relies on the same font height/metrics for each glyph", fontSize, fontPixelHeight));
```

Comment:
Didn't get here. If fontPixelHeight should be _always_ the same, why not set it once into GlyphManager and don't pass into ShapeText?

But is it true that font height should always be the same during the app's lifetime?

Reply:
Now generating and caching texts of the same size, and then scaling them only right before the rendering is easier and faster (the same approach was used in the previous drape implementation btw, all computed metrics had the same font size). 

However, producing the correct final/scaled size provides slightly more accurate metrics. Here it is left as a parameter for testing, and for possible future refactoring.

I've changed it to:
```
#ifdef DEBUG
  static int const fontSize = fontPixelHeight;
  ASSERT_EQUAL(fontSize, fontPixelHeight,
      ("Cache relies on the same font height/metrics for each glyph", fontSize, fontPixelHeight));
#endif

```

Reply:
Well, nit, but you can put "ethalon" size into class member for DEBUG assertions instead of static var.